src/parser/core/modules/ActionWindow/windows/BuffWindow.ts

```diff
@@ -47,15 +49,15 @@ export abstract class BuffWindow extends ActionWindow {
 		const buffFilter = filter<Event>()
 			.source(this.parser.actor.id)
 			.target(noneOf(pets))
-			.status(this.buffStatus.id)
+			.status(oneOf(ensureArray(this.buffStatus).map(s => s.id)))
 
 		this.addEventHook(buffFilter.type('statusApply'), this.startWindowAndTimeout)
 		this.addEventHook(buffFilter.type('statusRemove'), this.endWindowByStatus)
 	}
 
 	private startWindowAndTimeout(event: Events['statusApply']) {
 		this.onWindowStart(event.timestamp)
-		const duration = this.buffStatus.duration
+		const duration = _.max(ensureArray(this.buffStatus).map(s => s.duration))
```

Comment:
Since we know which status is being applied here, this should probably use the actual status rather than max duration of all possible.

```suggestion
		const duration = this.data.getStatus(event.status)?.duration
```